### ISSUE TITLE
`panic!` should not trigger `clippy::panic` in conditional expressions evaluated at compile-time

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#8358]
## Code

```rust
#![deny(clippy::panic)]

const _N: usize = 1;
const _: () = {
    if _N > 2 {
        panic!("Peek a boo");
    }
};
```

## Output

```
error: `panic` should not be present in production code
 --> src/lib.rs:6:9
  |
6 |         panic!("Peek a boo");
  |         ^^^^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![deny(clippy::panic)]
  |         ^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#panic

error: could not compile `playground` due to previous error
```

**Comment by c410-f3r** [comment]
Fixed by https://github.com/rust-lang/rust-clippy/pull/8592

Totally forgot about this issue and I guess there are a bunch of opened issues that are already solved
